src/Console/Application.php

```diff
@@ -49,11 +51,18 @@ public function __construct()
     {
         $this->filesystem = new Filesystem();
         $this->container = $this->getContainer();
-        $this->setDispatcher($this->container->get('event_dispatcher'));
 
         parent::__construct(self::APP_NAME, self::APP_VERSION);
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function setDispatcher(EventDispatcherInterface $dispatcher)
```

Comment:
Why did you move this one outside the constructor? It seems better to set a dispatcher then to overwrite thet set* method.